app/src/main/java/org/astraea/topic/ReplicaCollie.java

```diff
@@ -9,62 +9,265 @@
-                          + targetBrokers
+                          + argument.toBrokers
                           + " current: "
                           + currentBrokers
                           + " removed: "
-                          + args.fromBrokers);
-                var finalBrokers = new HashSet<>(keptBrokers);
-                finalBrokers.addAll(
-                    new ArrayList<>(availableBrokers).subList(0, numberOfMigratedReplicas));
-                result.put(tp, Map.entry(currentBrokers, finalBrokers));
+                          + argument.fromBrokers);
+                brokerMigrate.put(tp.getKey(), Map.entry(currentBrokers, argument.toBrokers));
+              } else {
+                brokerMigrate.put(tp.getKey(), Map.entry(currentBrokers, currentBrokers));
               }
             });
-    result.forEach(
+    return brokerMigrate;
+  }
+
+  static TreeMap<TopicPartition, Map.Entry<Set<String>, Set<String>>> checkMigratorPath(
+      TopicAdmin admin, Argument argument) {
+    TreeMap<TopicPartition, Map.Entry<Set<String>, Set<String>>> pathMigrate =
+        new TreeMap<>(
+            Comparator.comparing(TopicPartition::topic).thenComparing(TopicPartition::partition));
+    admin.replicas(argument.topics).entrySet().stream()
+        .filter(
+            tp ->
+                argument.topics.contains(tp.getKey().topic())
+                    && ((argument.partitions.isEmpty())
+                        || argument.partitions.contains(tp.getKey().partition())))
+        .collect(Collectors.toList())
+        .forEach(
+            (tp) -> {
+              Set<String> fromPath = new HashSet<>();
+              Set<String> toPath = new HashSet<>();
+              tp.getValue().stream()
+                  .filter(assignments -> argument.fromBrokers.contains(assignments.broker()))
+                  .forEach(
+                      assignments -> {
+                        var currentPath = assignments.path();
+                        fromPath.add(currentPath);
+                        if (argument.path != null) {
+                          if (!currentPath.equals(argument.path.iterator().next()))
+                            toPath.add(argument.path.iterator().next());
+                        } else {
+                          if (argument.fromBrokers.contains(argument.toBrokers.iterator().next())) {
+                            toPath.add(
+                                admin
+                                    .brokerFolders(argument.toBrokers)
+                                    .get(argument.toBrokers.iterator().next())
+                                    .stream()
+                                    .filter(p -> !currentPath.contains(p))
+                                    .collect(Collectors.toSet())
+                                    .iterator()
+                                    .next());
+                          } else {
+                            toPath.add("unknown");
```

Comment:
We should avoid using this magic word ...

Could you add a static member to replace the string?